docs/SelfDiagnostics.MD

```diff
@@ -0,0 +1,65 @@
+# [Advanced] How to enable self diagnostics for ApplicationInsights.Kubernetes
+
+## Enable self-diagnostics
+
+We understand ApplicationInsights.Kubernetes is far from perfect. Enabling self-diagnostics will help find [issues](https://github.com/microsoft/ApplicationInsights-Kubernetes/issues) easier. Please feel free to open issues and we will be glad to look into them.
+
+Application Insights for Kubernetes is instrumented with DiagnosticsSource. To see the logs, create a diagnostic source listener that listens to `Microsoft.ApplicationInsights.Kubernetes.Debugging.Inspect.Observer`. The observer(DiagnosticSource) will emits logs in different levels.
```

Comment:
OK, this is a good example of bad naming that makes things confusing.

Assuming `Inspect` becomes `XxxDiagnosticSource`, the property should be called `Listener`. You should then instruct the user to create an `Observer` that subscribes to the `Listener` to consume logs from the AI.K package